components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -686,6 +706,93 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         return consentPage;
     }
 
+    private static String getFilteredScopes(OAuth2Parameters params) throws OAuthSystemException {
+
+        try {
+            Set<String> allowedScopes = params.getScopes();
+            startTenantFlow(params.getTenantDomain());
```

Comment:
For which calls do we need the tenant flow started. If its just for the oAuthAdminService then would be better to scope the try-finally block to that particular code segment only